The patch has been taken from community and fixes skipNFSInHostResources handling of hrStorageTable.

The details can be found in the following location
https://sourceforge.net/p/net-snmp/code/ci/94ef3205ac323f6d11ee6b3317484e4eda746c3a/

--- net-snmp-5.4.1/agent/mibgroup/host/hr_storage.c	2016-03-23 04:42:08.681623040 -0700
+++ copy_net-snmp-5.4.1/agent/mibgroup/host/hr_storage.c	2016-03-23 05:11:17.386789330 -0700
@@ -524,12 +524,11 @@
 	    return NULL;
 
         store_idx = name[ HRSTORE_ENTRY_NAME_LENGTH ];
-        if (HRFS_entry &&
-            netsnmp_ds_get_boolean(NETSNMP_DS_APPLICATION_ID,
-                                   NETSNMP_DS_AGENT_SKIPNFSINHOSTRESOURCES) &&
-            Check_HR_FileSys_NFS())
-            return NULL;
         if (store_idx > NETSNMP_MEM_TYPE_MAX ) {
+	    if ( netsnmp_ds_get_boolean(NETSNMP_DS_APPLICATION_ID,
+                                        NETSNMP_DS_AGENT_SKIPNFSINHOSTRESOURCES) &&
+                 			Check_HR_FileSys_NFS())
+               	return NULL;  /* or goto try_next; */
 	    if (HRFS_statfs(HRFS_entry->HRFS_mount, &stat_buf) < 0) {
 		snmp_log_perror(HRFS_entry->HRFS_mount);
 		goto try_next;
